diff --git a/src/CircularDependencyException.php b/src/CircularDependencyException.php
index 201276f..b17496a 100644
--- a/src/CircularDependencyException.php
+++ b/src/CircularDependencyException.php
@@ -13,7 +13,7 @@ class CircularDependencyException extends \Exception
      * @param \Exception $previous
      * @param string[]   $nodes
      */
-    public function __construct($message = '', $code = 0, \Exception $previous = null, $nodes)
+    public function __construct($message, $code, ?\Exception $previous, $nodes)
     {
         parent::__construct($message, $code, $previous);
         array_pop($nodes);
diff --git a/src/ElementNotFoundException.php b/src/ElementNotFoundException.php
index 3adfd55..a387da8 100644
--- a/src/ElementNotFoundException.php
+++ b/src/ElementNotFoundException.php
@@ -21,7 +21,7 @@ class ElementNotFoundException extends \Exception
      * @param string     $source
      * @param string     $target
      */
-    public function __construct($message = '', $code = 0, \Exception $previous = null, $source, $target)
+    public function __construct($message, $code, ?\Exception $previous, $source, $target)
     {
         parent::__construct($message, $code, $previous);
         $this->source = $source;
